eureka-client-jersey2/src/main/java/com/netflix/discovery/shared/transport/jersey2/Jersey2ApplicationClientFactory.java

```diff
@@ -162,12 +163,12 @@ public void filter(ClientRequestContext requestContext) {
 
             JerseyClient jersey2Client = (JerseyClient) clientBuilder.build();
 
-            Map<String, String> additionalHeaders = new HashMap<>();
+            MultivaluedMap<String, Object> additionalHeaders = new MultivaluedHashMap<>();
             if (allowRedirect) {
-                additionalHeaders.put(HTTP_X_DISCOVERY_ALLOW_REDIRECT, "true");
+                additionalHeaders.put(HTTP_X_DISCOVERY_ALLOW_REDIRECT, Arrays.<Object>asList("true"));
```

Comment:
Should be able to use add instead of put to avoid the list creation.

http://docs.oracle.com/javaee/7/api/javax/ws/rs/core/MultivaluedMap.html#add-K-V-